fridgify/lib/data/content_repository.dart

```diff
@@ -0,0 +1,175 @@
+
+import 'dart:convert';
+
+import 'package:fridgify/data/item_repository.dart';
+import 'package:fridgify/data/repository.dart';
+import 'package:fridgify/data/store_repository.dart';
+import 'package:fridgify/exception/failed_to_add_content_exception.dart';
+import 'package:fridgify/exception/failed_to_fetch_api_token_exception.dart';
+import 'package:fridgify/exception/failed_to_fetch_content_exception.dart';
+import 'package:fridgify/model/content.dart';
+import 'package:fridgify/model/fridge.dart';
+import 'package:fridgify/model/item.dart';
+import 'package:logger/logger.dart';
+import 'package:shared_preferences/shared_preferences.dart';
+import 'package:http/http.dart' as http;
+
+class ContentRepository implements Repository<Content> {
+
+  Logger logger = Logger();
+
+  Fridge fridge;
+  SharedPreferences pref = Repository.sharedPreferences;
+  ItemRepository itemRepository = ItemRepository();
+
+  Map<int, Content> contents = Map();
+
+  var contentApi;
+
+  ContentRepository(this.pref, this.fridge) {
+    contentApi = "${Repository.baseURL}fridge/content/${this.fridge.fridgeId}/";
+  }
+
+  @override
+  Future<int> add (Content content) async {
+    var token = getToken();
+    var body = jsonEncode({
+      "name": content.item.name, "description": content.item.description,
+      "buy_date": "2020-04-04", "expiration_date": content.expirationDate,
+      "amount": content.amount,
+      "unit": content.unit, "store": content.item.store.name,
+    });
+    logger.i('ContentRepository => Requesting $contentApi with $body}');
```

Comment:
I guess you missed the '}' or has that curly bracket any use ?

Reply:
Thanks for noticing that, I removed the forgotten '}'.

I also implemented a getHeader function as seen in 50bb59b for easier header usage/generation.